src/common/transactions/postcondition-utils.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { handlePostConditions } from '@common/transactions/postcondition-utils';
+import {
+  createAssetInfo,
+  hexToCV,
+  makeContractNonFungiblePostCondition,
+  makeStandardNonFungiblePostCondition,
+  NonFungibleConditionCode,
+  parsePrincipalString,
+} from '@stacks/transactions';
+
+const SENDER_ADDRESS = 'ST2PABAF9FTAJYNFZH93XENAJ8FVY99RRM4DF2YCW';
+
+describe(handlePostConditions.name, function () {
+  it('should not modify a post condition where the principal is a contract', () => {
+    const postCondition = makeContractNonFungiblePostCondition(
+      SENDER_ADDRESS,
+      'btc-nft-swap-v1',
+      NonFungibleConditionCode.DoesNotOwn,
+      createAssetInfo(
+        'ST248HH800501WYSG7Z2SS1ZWHQW1GGH85Q6YJBCC',
+        'passive-blue-marmot',
+        'layer-nft'
+      ),
+      hexToCV('0x0100000000000000000000000000000003')
+    );
+    const transformedPostCondition = handlePostConditions(
+      [postCondition],
+      SENDER_ADDRESS,
+      SENDER_ADDRESS
+    );
+
+    expect(transformedPostCondition[0]).toEqual(postCondition);
+  });
+
+  it('should not modify a post condition addresses are all the same', () => {
```

Comment:
Not sure this statement reads correctly?

Reply:
yeah should be something like
```suggestion
  it('should not modify a post condition when all addresses are the same (sender, current address, and postcondition principal)', () => {
```